src/components/Calendar/Calendar.tsx

```diff
@@ -0,0 +1,173 @@
+"use client";
+
+import React, { useState } from "react";
+import FullCalendar from "@fullcalendar/react";
+import timeGridPlugin from "@fullcalendar/timegrid";
+import dayGridPlugin from "@fullcalendar/daygrid";
+import interactionPlugin, { DateClickArg } from "@fullcalendar/interaction";
+import { EventClickArg } from "@fullcalendar/core";
+import styled from "styled-components";
+import { Dialog } from "@mui/material";
+
+interface CalendarProps {
+    initialEvents: CalendarEvent[];
+    view?: string;
+    editable?: boolean;
+    handleDateClick?: (info: DateClickArg) => void;
+}
+
+export interface CalendarEvent {
+    id: string;
+    title: string;
+    start: Date;
+    end: Date;
+    description?: string;
+    daysOfWeek?: number[];
+    startRecur?: string;
+    endRecur?: string;
+    backgroundColor?: string;
+    borderColor?: string;
+    textColor?: string;
+    allDay?: boolean;
+}
+
+const StyledDialog = styled(Dialog)`
+    // root
+    & > div {
+        // container
+        > div {
+            //Paper
+            width: min(50%, 600px);
+            border-radius: 2rem;
+            padding: 1.5rem;
+            text-align: center;
+        }
+    }
+`;
+
+const StyledCancelButton = styled.button`
+    position: absolute;
+    right: 20px;
+    top: 20px;
+`;
+
+const ModalInner = styled.div`
+    display: flex;
+    flex-direction: column;
+    row-gap: 1rem;
+`;
+
+const CalendarStyling = styled.div`
+    .fc {
+        background-color: white;
+        border-radius: 2rem;
+        padding: 1.5rem;
+        text-align: center;
+    }
+
+    .fc .fc-toolbar-title {
+        font-size: 1.5rem;
+    }
+
+    .fc .fc-toolbar-chunk {
+        margin: 0 0.5rem;
+    }
+
+    .fc .fc-button {
+        height: max(3vmin, 30px);
+        font-size: max(1.5vmin, 15px);
+        text-align: center;
+        vertical-align: middle;
+        padding: 0 10px 0 10px;
+    }
+
+    .fc .fc-icon {
+        font-size: max(1.5vmin, 15px);
+    }
+
+    .fc .fc-button-primary {
+        background-color: ${(props) => props.theme.fillColor};
+        border-color: ${(props) => props.theme.fillColor};
+        color: black;
+    }
+
+    .fc .fc-button-primary:hover {
```

Comment:
some of these CSS names are not immediately obvious what they are - can you add a small comment inline about what they are if not absolutely obvious?

Reply:
still unsure on what a 'primary' button is, why you need to separate this out, or if `.fc-button` is 'all visible buttons'

also, can the `:hover` be embedded in `button-primary` with an `&`? thanks :)